### PR TITLE
Tier 3 naming convention

### PR DESCRIPTION
**Pull request by vidyambala**
# Description

Update 3 workload resource names consistent with other MLZ resource names

## Issue reference

The issue this PR will close: #612 

## Checklist

Please make sure you've completed the relevant tasks for this PR out of the following list:

* [x] All acceptance criteria in the backlog item are met
* [x] The documentation is updated to cover any new or changed features
* [x] Manual tests have passed
* [x] Relevant issues are linked to this PR

### CODE REVIEW

File: src/bicep/examples/newWorkload/README.md [hunk 1]
```diff
@@ -16,19 +16,19 @@ The docs on Azure virtual networking:  <https://docs.microsoft.com/en-us/azure/v
 ## Pre-requisites
```
Comment (glennmusa) [inline src/bicep/examples/newWorkload/README.md:16]: I made updates to the README. Let me know if these are changes are correct by resolving this conversation.

File: src/bicep/examples/newWorkload/newWorkload.bicep [hunk 2]
```diff
@@ -1,30 +1,55 @@
...
+    - resourcePrefix
+
+*/
+
+// REQUIRED PARAMETERS
+
+@minLength(3)
+@maxLength(10)
+@description('A prefix, 3 to 10 characters in length, to append to resource names (e.g. "dev", "test", "prod", "mlz"). It defaults to "mlz".')
+param resourcePrefix string
```
Comment (glennmusa) [inline src/bicep/examples/newWorkload/newWorkload.bicep:18]: I updated this to match the convention used in `mlz.bicep`.  Let me know if these are changes are correct by resolving this conversation.

https://github.com/Azure/missionlz/blob/199b986c770d0d331b6d99fce39d9b9807af2e28/src/bicep/mlz.bicep#L16-L19

File: src/bicep/examples/newWorkload/newWorkload.bicep [hunk 3]
```diff
@@ -67,32 +125,32 @@ module spokeNetwork '../../modules/spokeNetwork.bicep' = {
...
-    subnetName: subnetName
+    subnetName: workloadSubnetName
     subnetAddressPrefix: subnetAddressPrefix
     subnetServiceEndpoints: subnetServiceEndpoints
   }
 }
 
 module workloadVirtualNetworkPeerings '../../modules/spokeNetworkPeering.bicep' = {
-  name: take('${workloadName}--VNetPeerings', 64)
+  name: take('${workloadName}-to-hub-vnet-peering', 64)
```
Comment (glennmusa) [inline src/bicep/examples/newWorkload/newWorkload.bicep:153]: I updated this to match the convention used in `hubNetworkPeerings.bicep`.  Let me know if these are changes are correct by resolving this conversation.

https://github.com/Azure/missionlz/blob/199b986c770d0d331b6d99fce39d9b9807af2e28/src/bicep/modules/hubNetworkPeerings.bicep#L5

File: src/bicep/examples/newWorkload/newWorkload.bicep [hunk 4]
```diff
@@ -105,7 +163,7 @@ module workloadVirtualNetworkPeerings '../../modules/spokeNetworkPeering.bicep'
 
 module hubToWorkloadVirtualNetworkPeering './modules/hubNetworkPeering.bicep' = {
   scope: subscription(hubSubscriptionId)
-  name: 'hubToWorkloadVirtualNetworkPeering'
+  name: take('hub-to-${workloadName}-vnet-peering', 64)
```
Comment (glennmusa) [inline src/bicep/examples/newWorkload/newWorkload.bicep:166]: I updated this to match the convention used in `hubNetworkPeerings.bicep`.  Let me know if these are changes are correct by resolving this conversation.

https://github.com/Azure/missionlz/blob/199b986c770d0d331b6d99fce39d9b9807af2e28/src/bicep/modules/hubNetworkPeerings.bicep#L5

**Review by glennmusa — COMMENTED** [review]
I made some updates to follow conventions that exist elsewhere in the repo and to the docs. Let me know if there's things I've overlooked!👍

File: src/bicep/examples/newWorkload/newWorkload.bicep (hunk 2 above)
Comment (vidyambala) [reply src/bicep/examples/newWorkload/newWorkload.bicep:18]: perfect.

File: src/bicep/examples/newWorkload/newWorkload.bicep (hunk 4 above)
Comment (vidyambala) [reply src/bicep/examples/newWorkload/newWorkload.bicep:166]: thanks for the update

**Review by glennmusa — APPROVED** [review]

**Pull request merged**